pkg/store/bucket.go

```diff
@@ -1469,6 +1469,7 @@ func (b *bucketBlock) readChunkRange(ctx context.Context, seq int, off, length i
 	}
 
 	if seq < 0 || seq >= len(b.chunkObjs) {
+		b.chunkPool.Put(c)
 		return nil, errors.Errorf("unknown segment file for index %d", seq)
 	}
```

Comment:
What about moving this if block to the start of the method? Then we don't need to allocate from the chunk pool.

Reply:
Sure. It's proposed in the commit message. I wanted to highlight the imbalance first.

Reply:
Updated.